developer/reference/articles/tooling/blade-cli/08-extending-blade-cli/04-installing-new-extensions-for-blade-cli.markdown

```diff
@@ -0,0 +1,70 @@
+---
+header-id: installing-new-extensions-for-blade-cli
+---
+
+# Installing New Extensions for Blade CLI
+
+[TOC levels=1-4]
+
+After you've created a new extension for Blade CLI, you must install it so it's
+available for use. You can learn how to create
+[custom commands](/docs/7-2/reference/-/knowledge_base/r/creating-custom-commands-for-blade-cli)
+and
+[custom project templates](/docs/7-2/reference/-/knowledge_base/r/creating-custom-project-templates-for-blade-cli)
+in their respective articles.
+
+When Blade CLI starts, it looks in the user's `${user.home}/.blade/extensions`
+folder for any JAR files. All JAR files are searched to see if they contain
+valid Blade extensions. You'll learn how to install new extensions next.
+
+## Installing a New Extension
+
+To install an extension, you must move the extension JAR to the user's
+`${user.home}/.blade/extensions` folder. You can do this automatically from
+Blade CLI. There are three ways to install a new Blade extension:
+
+**Local JAR file:**
+
+```bash
+blade extension install /path/to/my_extension.JAR
+```
+
+**Remote JAR file indicated by a URL:**
+
+```bash
+blade extension install https://repository.liferay.com/public/group/com/liferay/blade/com.liferay.blade.extension/1.0.0/com.liferay.blade.extension-1.0.0.JAR
+```
+
+**Github repository that contains an extension:**
+
+```bash
+blade extension install https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-command
+```
+
```

Comment:
Maybe we should make a note here that the third choice, installing directly from a github url, will attempt to build the source code of the project and then install that as a blade extension jar.

Reply:
For the remote JAR by URL (second choice) and Github repo URL (third choice) methods, are the projects downloaded locally before they're installed? Where would that be?

Reply:
I just looked at that code to confirm, and currently for github links, the github project is downloaded as a zip file into a temporary directory, unzipped, and Blade builds it as a gradle project.

While reviewing the code, I found it appears that we **don't** currently support pointing the url to a jar file directly on the internet, like this `blade extension install http://some.jar` - or any http links that aren't github.

Though, I think this is a feature we should have. 

There is a workaround: Downloading the jar file directly and then running `blade extension install /path/to/file.jar`, so there is currently an extra step in the process.

Reply:
Very good catch! I'll update these docs to be consistent with our current offering.